### ISSUE TITLE
About Testing

### ISSUE THREAD

**Issue by AmingWu** [HaozheQi/P2B#9]
Where are the testing results?
![QQ截图20200617142630](https://user-images.githubusercontent.com/38445175/84863038-d4d4d880-b0a6-11ea-8cae-81d331738505.png)

**Comment by HaozheQi** [comment]
Hi, @AmingWu, actually 'succ/prec' in the picture is the result, but there may be something wrong about your result. The true result is about 56.2/72.8. you may need to check the input settings of the code.
![84863038-d4d4d880-b0a6-11ea-8cae-81d331738505](https://user-images.githubusercontent.com/32994368/84867140-90f2bc80-b02f-11ea-9e02-e217839b0367.png)

**Comment by AmingWu** [comment]
I find the testing data does not include label_02. The label_02 is only for training. Is it correct?

**Comment by AmingWu** [comment]
![QQ截图20200628125037](https://user-images.githubusercontent.com/38445175/85938244-13d41b00-b93e-11ea-95df-5d8b6180b363.png)

**Comment by AmingWu** [comment]
data_tracking_label_2 only contains the training data. And it does not contain the test data.

**Comment by HaozheQi** [comment]
Yes, the KITTI tracking dataset only gives the label of training set. So we only used its training set to train and test our P2B, which was split as follows: scenes 0-16 for training, 17-18 for validation, and 19-20 for testing.

**Comment by AmingWu** [comment]
![QQ截图20200628185028](https://user-images.githubusercontent.com/38445175/85945426-44cd4380-b970-11ea-9697-d57aa4dffd00.png)
I retrain the code. However, I still could not reproduce your results.

**Comment by AmingWu** [comment]
I use the default settings of your code.

**Comment by HaozheQi** [comment]
The result 6.1/3.9 indicates the tracker can not give a right prediction at all, I think there may still exist some errors in your code. Can you get a right result with the model I trained?

**Comment by AmingWu** [comment]
![QQ截图20200629070221](https://user-images.githubusercontent.com/38445175/85960524-aa541b00-b9d6-11ea-9bad-eb1eb1c2534c.png)
This is the training process. Is it correct?

**Comment by AmingWu** [comment]
![QQ截图20200629141709](https://user-images.githubusercontent.com/38445175/85979932-00937f00-ba14-11ea-8146-8a1ba4db05b1.png)
The training is right?

**Comment by HaozheQi** [comment]
The output of your terminal seems to be correct. I'm not sure why your test result is wrong.

**Comment by AmingWu** [comment]
I use the default settings. And I do not modify the code. The dataset is correct. But, I do not reproduce your result.

**Comment by AmingWu** [comment]
Could you share your trained model?

**Comment by HaozheQi** [comment]
The model trained for car is already included in this repository, and I'm working on the release of other categories.

**Comment by AmingWu** [comment]
OK, I have test your model. I find the result is still wrong.

**Comment by HaozheQi** [comment]
You may check the code related to the test stage, I think your training stage has no error now.

**Comment by AmingWu** [comment]
dataset_Test = SiameseTest(
            input_size=1024,
            path='/home/room/wam/P2B/trackingdata/testing',
            split='Test',
            category_name='Car',
            offset_BB=0,
            scale_BB=1.25)

I only modify the path from opt.data_dir to '/home/room/wam/P2B/trackingdata/testing',

**Comment by HaozheQi** [comment]
The path of training and test should be the same, does this matter?

**Comment by AmingWu** [comment]
Thank you very much. I have reproduced your results. The path of training and test should be the same.

**Comment by AmingWu** [comment]
If I want to run other categories, I only need to modify parser.add_argument('--category_name', type=str, default = 'Car',  help='Object to Track (Car/Pedetrian/Van/Cyclist)')?

**Comment by zhanghm1995** [comment]
@AmingWu @HaozheQi, Hello, I wonder how did you final solve this problem, I run the following scripts:
```python
python test_tracking.py --data_dir=/mnt/study/Datasets/KITTI/tracking/training
```
And also got the result like yours:
```shell
Test 1: Time 0.030s (it:0.031s) Data:0.409s (it:0.518s), Succ/Prec:6.6/4.4: 100%
```
And I used the KITTI training data and the default offline `netR_36.pth` model file.

**Comment by zjtgit** [comment]
> @AmingWu @HaozheQi, Hello, I wonder how did you final solve this problem, I run the following scripts:
> 
> ```python
> python test_tracking.py --data_dir=/mnt/study/Datasets/KITTI/tracking/training
> ```
> 
> And also got the result like yours:
> 
> ```shell
> Test 1: Time 0.030s (it:0.031s) Data:0.409s (it:0.518s), Succ/Prec:6.6/4.4: 100%
> ```
> 
> And I used the KITTI training data and the default offline `netR_36.pth` model file.

I run the test script in pycharm, and I got the same result as yours. I am sure the data which I used to test is KITTI training sequence, but it's still give me a wrong result.
```shell
Test 1: Time 0.037s (it:0.041s) Data:0.436s (it:0.493s), Succ/Prec:6.6/4.4: 100%|██████████| 6424/6424 [03:56<00:00, 27.15it/s]
```

**Comment by wangkai602058765** [comment]
> Thank you very much. I have reproduced your results. The path of training and test should be the same.
I want to know how to use the KITTI datasets to test.It seems that i am failed to load the dataset.


![屏幕截图 2021-07-05 105800](https://user-images.githubusercontent.com/42930372/124411881-f5d69d00-dd7f-11eb-9c56-1a37704957f2.png)

**Comment by wangkai602058765** [comment]
> OK, I have test your model. I find the result is still wrong.

![屏幕截图 2021-07-05 105800](https://user-images.githubusercontent.com/42930372/124411974-1e5e9700-dd80-11eb-9233-8e34175e008e.png)
hi,i want to know how to use the KITTI dataset. It seems that i am failed to load the dataset.@AmingWu
